src/main/java/seedu/bigpp/command/commoncommand/BackCommand.java

```diff
@@ -1,12 +1,32 @@
 package seedu.bigpp.command.commoncommand;
-
 import seedu.bigpp.command.Command;
+import seedu.bigpp.ui.UI;
+import seedu.bigpp.ui.UIState;
 
 public class BackCommand extends Command {
 
     @Override
     public String executeCommand() {
-        // TODO Auto-generated method stub
-        return "";
+        int previousStateIndex = UI.visitedMenus.size() - 2;
```

Comment:
Instead of using previous state, since all the menu's only have one valid previous state, we can instead get the current menu state and transition to that valid previous state